clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -97,7 +101,23 @@ struct LoweringPreparePass : public LoweringPrepareBase<LoweringPreparePass> {
   /// -----------
 
   clang::ASTContext *astCtx;
-  void setASTContext(clang::ASTContext *c) { astCtx = c; }
+  std::shared_ptr<::cir::LoweringPrepareCXXABI> cxxABI;
+
+  void setASTContext(clang::ASTContext *c) {
+    astCtx = c;
+    switch (c->getCXXABIKind()) {
+    case clang::TargetCXXABI::GenericItanium:
+    case clang::TargetCXXABI::GenericAArch64:
+    case clang::TargetCXXABI::AppleARM64:
+      // TODO: this isn't quite right, clang uses AppleARM64CXXABI which
+      // inherits from ARMCXXABI. We'll have to follow suit.
+      cxxABI.reset(::cir::LoweringPrepareCXXABI::createItaniumABI());
```

Comment:
Not sure I understand what's going on here, please elaborate.

Reply:
This piece of code is just copied from here (where `CIRGenCXXABI` objects are created before CIRGen starts):

https://github.com/llvm/clangir/blob/b7e1c765db62c8e0fdb52a0644794db940d0db08/clang/lib/CIR/CodeGen/CIRGenItaniumCXXABI.cpp#L360-L376